### PR TITLE
test/lint: Always fetch the target branch if it cannot be found.

### PR DESCRIPTION
**Pull request by markylaing**
Previously this was only performed when the GITHUB_BEFORE environment variable was set.

### CODE REVIEW

File: test/lint/golangci.sh [hunk 1]
```diff
@@ -29,6 +26,11 @@ if [ -z "${target_branch}" ]; then
   return
 fi
 
+if ! git show-ref --quiet "refs/heads/${target_branch}" --quiet >/dev/null 2>&1; then
+    # Attempt to fetch the reference (Github uses a shallow clone).
+    git fetch -q origin "${target_branch}" || true
```
Comment (tomponline) [inline test/lint/golangci.sh:31]: should we still have the `|| true` bit here?

Comment (markylaing) [reply test/lint/golangci.sh:31]: Probably not! :facepalm:

**Comment by markylaing** [comment]
@tomponline this seems to be working now. I've changed the script to abort if it doesn't find a revision to check against, otherwise no lint errors will be raised.

I am quite confused as to how we haven't come across this before. It's possible that we just didn't notice the warning and nothing has been linted with golangci for quite some time :cry:

**Comment by markylaing** [comment]
> @tomponline this seems to be working now. I've changed the script to abort if it doesn't find a revision to check against, otherwise no lint errors will be raised.
> 
> I am quite confused as to how we haven't come across this before. It's possible that we just didn't notice the warning and nothing has been linted with golangci for quite some time 😢

Indeed this is from the next PR that changed any Go code (https://github.com/canonical/lxd/pull/12649), it was in December: https://github.com/canonical/lxd/actions/runs/7164995944/job/19506095032#step:14:19

**Comment by tomponline** [comment]
I'm glad I opportunistically noticed and mentioned it then :)

**Review by tomponline — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**